src/main/java/org/chtracker/dao/report/Attack.java

```diff
@@ -1,16 +1,66 @@
 package org.chtracker.dao.report;
 
 import java.time.LocalDateTime;
-
+import java.util.ArrayList;
+import java.util.Collection;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.ForeignKey;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import org.chtracker.dao.DataConfiguration;
 import org.chtracker.dao.profile.Patient;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+@Entity
+@Table(
+		schema = DataConfiguration.REPORT_SCHEMA_NAME, 
+		uniqueConstraints = { @UniqueConstraint(
+				name = "attack_uniq",
+				columnNames = { "started", "patient_id" }
+				) 
+		})
 public class Attack {
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private int id;
+
+	@NotNull
 	private LocalDateTime started;
 	private LocalDateTime stopped;
-	private final Patient patient;
+	
+	@JsonIgnore
```

Comment:
Actually that's the reason, why I prefer to define explicit DTOs for presentation layer.

Reply:
Like we discussed yesterday. 
We can write code that efficiently used bandwidth but it will require large code base to maintain.
Or we can use model classes as cross cutting concerns. I'd start with this approach and switch to dedicated DTO or Projections if we face problems.